### PR TITLE
Add upsert capabilities

### LINKED ISSUE #5: Backend should accept inser or update in a single call

### PR DESCRIPTION
**Pull request by marcopeg**
closes #5

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -24,11 +24,16 @@ server.get("/:uname", async (request, reply) => {
...
+  const sql = `
+    INSERT INTO cv_data VALUES ($1, $2)
+    ON CONFLICT ON CONSTRAINT cv_data_pkey DO
+    UPDATE SET data = EXCLUDED.data
+    RETURNING data
+  `; 
+  const values = [request.params.uname, request.body];
   const result = await client.query(sql, values);
-  reply.send(result);
+  reply.send(result.rows[0].data);
```
Comment (VernitaJ) [inline src/index.js:36]: Looking good

File: src/index.js [hunk 2]
```diff
@@ -24,11 +24,16 @@ server.get("/:uname", async (request, reply) => {
...
 
-server.post("/", async (request, reply) => {
-  const sql = "INSERT INTO todos (text) VALUES ($1)";
-  const values = [request.body.text];
+server.post("/:uname", async (request, reply) => {
+  const sql = `
+    INSERT INTO cv_data VALUES ($1, $2)
+    ON CONFLICT ON CONSTRAINT cv_data_pkey DO
+    UPDATE SET data = EXCLUDED.data
+    RETURNING data
```
Comment (marcopeg-dev) [inline src/index.js:32]: Oh nice query, i should save it for later

File: src/index.js [hunk 3]
```diff
@@ -24,11 +24,16 @@ server.get("/:uname", async (request, reply) => {
   reply.send(result.rows[0].data);
 });
 
-server.post("/", async (request, reply) => {
-  const sql = "INSERT INTO todos (text) VALUES ($1)";
-  const values = [request.body.text];
+server.post("/:uname", async (request, reply) => {
+  const sql = `
+    INSERT INTO cv_data VALUES ($1, $2)
+    ON CONFLICT ON CONSTRAINT cv_data_pkey DO
```
Comment (NicolaWinterson) [inline src/index.js:30]: Didn't know about this. Cool!

**Review by marcopeg-dev — APPROVED** [review]
👍

**Pull request merged**